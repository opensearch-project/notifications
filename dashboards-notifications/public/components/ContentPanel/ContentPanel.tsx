/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiHorizontalRule,
  EuiPanel,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';

interface ContentPanelProps {
  title?: string;
  titleSize?: 'xxxs' | 'xxs' | 'xs' | 's' | 'm' | 'l';
  total?: number;
  bodyStyles?: object;
  panelStyles?: object;
  horizontalRuleClassName?: string;
  actions?: React.ReactNode | React.ReactNode[];
  children: React.ReactNode | React.ReactNode[];
}

const ContentPanel: React.SFC<ContentPanelProps> = ({
  title = '',
  titleSize = 'l',
  total = undefined,
  bodyStyles = {},
  panelStyles = {},
  horizontalRuleClassName = '',
  actions,
  children,
}) => (
  <EuiPanel style={{ ...panelStyles }}>
    <EuiFlexGroup
      style={{ padding: '0px 0px' }}
      justifyContent="spaceBetween"
      alignItems="center"
    >
      <EuiFlexItem>
        <EuiTitle size={titleSize}>
          <h3>
            {title}
            {total !== undefined ? (
              <span
                style={{ color: '#9f9f9f', fontWeight: 300 }}
              >{` (${total})`}</span>
            ) : null}
          </h3>
        </EuiTitle>
      </EuiFlexItem>
      {actions ? (
        <EuiFlexItem grow={false}>
          <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
            {Array.isArray(actions) ? (
              (actions as React.ReactNode[]).map(
                (action: React.ReactNode, idx: number): React.ReactNode => (
                  <EuiFlexItem key={idx}>{action}</EuiFlexItem>
                )
              )
            ) : (
              <EuiFlexItem>{actions}</EuiFlexItem>
            )}
          </EuiFlexGroup>
        </EuiFlexItem>
      ) : null}
    </EuiFlexGroup>

    <EuiHorizontalRule margin="s" className={horizontalRuleClassName} />

    <div style={{ padding: '0px 10px', ...bodyStyles }}>{children}</div>
  </EuiPanel>
);

export default ContentPanel;
