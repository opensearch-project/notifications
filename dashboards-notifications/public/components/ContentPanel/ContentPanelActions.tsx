/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';

interface ContentPanelActionsProps {
  actions: {
    component: React.ReactNode;
    flexItemProps?: object;
  }[];
}

const ContentPanelActions: React.SFC<ContentPanelActionsProps> = ({
  actions,
}) => (
  <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
    {actions.map(({ component, flexItemProps = {} }, index) => {
      return (
        <EuiFlexItem {...flexItemProps} grow={false} key={index}>
          {component}
        </EuiFlexItem>
      );
    })}
  </EuiFlexGroup>
);

export default ContentPanelActions;
