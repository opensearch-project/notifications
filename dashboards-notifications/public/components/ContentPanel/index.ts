/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import ContentPanel from './ContentPanel';
import ContentPanelActions from './ContentPanelActions';

export { ContentPanel, ContentPanelActions };
