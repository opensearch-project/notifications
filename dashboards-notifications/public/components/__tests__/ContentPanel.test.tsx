/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import ContentPanel from '../ContentPanel/ContentPanel';

describe('<ContentPanel /> spec', () => {
  it('renders the component', () => {
    const { container } = render(
      <ContentPanel title="Testing" actions={<div>one</div>}>
        <div>Testing ContentPanel</div>
      </ContentPanel>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders actions', () => {
    const { getByText } = render(
      <ContentPanel title="Testing" actions={[<div>one</div>, <div>two</div>]}>
        <div>Testing ContentPanel</div>
      </ContentPanel>
    );
    getByText('one');
    getByText('two');
  });

  it('renders with empty actions', () => {
    const { container } = render(
      <ContentPanel title="Testing">
        <div>Testing ContentPanel</div>
      </ContentPanel>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
