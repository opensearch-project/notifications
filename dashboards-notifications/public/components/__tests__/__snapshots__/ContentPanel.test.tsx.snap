// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentPanel /> spec renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 0px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--large"
      >
        Testing
      </h3>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div>
            one
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
  />
  <div
    style="padding: 0px 10px;"
  >
    <div>
      Testing ContentPanel
    </div>
  </div>
</div>
`;

exports[`<ContentPanel /> spec renders with empty actions 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 0px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--large"
      >
        Testing
      </h3>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
  />
  <div
    style="padding: 0px 10px;"
  >
    <div>
      Testing ContentPanel
    </div>
  </div>
</div>
`;
